Source/Core/Core/AchievementManager.cpp

```diff
@@ -895,6 +895,20 @@ const AchievementManager::NamedIconMap& AchievementManager::GetChallengeIcons()
   return m_active_challenges;
 }
 
+void AchievementManager::DoState(PointerWrap& p)
+{
+  int size = 0;
+  if (!p.IsReadMode())
+    size = rc_runtime_progress_size(&m_runtime, nullptr);
+  p.Do(size);
+  auto buffer = std::make_unique<u8>(size);
```

Comment:
```suggestion
  auto buffer = std::make_unique<u8[]>(size);
```

Otherwise this is allocating a single u8 with the value of size